packages/llamaindex/src/indices/vectorStore/index.ts

```diff
@@ -112,9 +113,20 @@ export class VectorStoreIndex extends BaseIndex<IndexDict> {
 
     if (options.nodes) {
       // If nodes are passed in, then we need to update the index
-      await index.buildIndexFromNodes(options.nodes, {
-        logProgress: options.logProgress,
-      });
+      try {
+        await index.buildIndexFromNodes(options.nodes, {
+          logProgress: options.logProgress,
+        });
+      } catch (error) {
+        // clean up doc store when generating embeddings fails
+        const { unusedDocs } = await classify(index.docStore, options.nodes);
+        for (const docId of unusedDocs) {
+          await index.docStore.deleteDocument(docId, false);
```

Comment:
@thucpn i think this doesn't cleanup a general call of `insertNodes` (`buildIndexFromNodes` is a synonym). we should catch failed calls of `getNodeEmbeddingResults` and then do a clean up (if necessary, see below)